### PR TITLE
tools: add a benchmark tool for hearbeat

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
To benchmark the performance of handing heartbeat. 

### What is changed and how it works?
example:
```
❯ ./heartbeat-bench
ClusterID: 6748798567986197874
bootstrapped
finish put stores

--------- Bench heartbeat (Round 1) ----------
store 1 finish heartbeat, cost time: 42.580442683s
store 9 finish heartbeat, cost time: 42.620485179s
store 19 finish heartbeat, cost time: 42.863172835s
store 18 finish heartbeat, cost time: 43.035728396s
store 20 finish heartbeat, cost time: 43.0527188s
store 6 finish heartbeat, cost time: 43.059825617s
store 10 finish heartbeat, cost time: 43.112139272s
store 13 finish heartbeat, cost time: 43.115090363s
store 3 finish heartbeat, cost time: 43.11638077s
store 5 finish heartbeat, cost time: 43.120537883s
store 12 finish heartbeat, cost time: 43.132455427s
store 16 finish heartbeat, cost time: 43.13932419s
store 14 finish heartbeat, cost time: 43.147619339s
store 4 finish heartbeat, cost time: 43.151354573s
store 2 finish heartbeat, cost time: 43.160081243s
store 8 finish heartbeat, cost time: 43.263776622s
store 7 finish heartbeat, cost time: 43.275595781s
store 11 finish heartbeat, cost time: 43.371469709s
store 15 finish heartbeat, cost time: 43.409647719s
store 17 finish heartbeat, cost time: 43.673221641s

Summary:
  Total:	43.6778 secs.
  Slowest:	0.8482 secs.
  Fastest:	0.0000 secs.
  Average:	0.0009 secs.
  Stddev:	0.0094 secs.
  Requests/sec:	22895.3795

Response time histogram:
  0.0000 [1]	|
  0.0848 [996730]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.1696 [3023]	|
  0.2545 [223]	|
  0.3393 [21]	|
  0.4241 [2]	|
  0.5089 [0]	|
  0.5937 [0]	|
  0.6786 [0]	|
  0.7634 [19]	|
  0.8482 [1]	|

Latency distribution:
  10% in 0.0000 secs.
  25% in 0.0000 secs.
  50% in 0.0000 secs.
  75% in 0.0000 secs.
  90% in 0.0000 secs.
  95% in 0.0000 secs.
  99% in 0.0006 secs.
  99.9% in 0.1058 secs.

Sample in one second (unix latency throughput):
UNIX-SECOND,MIN-LATENCY-MS,AVG-LATENCY-MS,MAX-LATENCY-MS,AVG-THROUGHPUT
1571327127,788ns,323.642µs,106.739947ms,25544
1571327128,781ns,747.551µs,130.067004ms,27126
1571327129,782ns,748.791µs,150.723973ms,26414
1571327130,784ns,776.141µs,143.202947ms,25636
1571327131,783ns,815.447µs,174.68032ms,24621
1571327132,778ns,853.462µs,142.031431ms,23793
1571327133,779ns,760.395µs,141.936154ms,25882
1571327134,784ns,798.214µs,162.616604ms,25009
1571327135,792ns,878.417µs,230.237056ms,23017
1571327136,781ns,866.963µs,184.986608ms,22656
1571327137,782ns,716.423µs,118.479538ms,28183
1571327138,775ns,980.729µs,268.449372ms,20249
1571327139,782ns,760.776µs,134.315416ms,26076
1571327140,786ns,932.908µs,299.566858ms,21413
1571327141,777ns,758.951µs,126.164672ms,26464
1571327142,775ns,863.657µs,152.634238ms,23502
1571327143,781ns,1.017866ms,231.457625ms,19473
1571327144,782ns,796.898µs,133.192405ms,25151
1571327145,782ns,1.072382ms,278.954867ms,19667
1571327146,780ns,796.124µs,155.28796ms,23529
1571327147,782ns,849.546µs,151.18481ms,23326
1571327148,779ns,801.438µs,139.823777ms,24815
1571327149,772ns,1.279535ms,291.934335ms,15683
1571327150,784ns,871.513µs,181.738208ms,23114
1571327151,778ns,938.72µs,158.016393ms,21316
1571327152,777ns,748.131µs,141.335129ms,26658
1571327153,784ns,1.262454ms,393.043323ms,15586
1571327154,778ns,723.609µs,140.224914ms,27628
1571327155,776ns,870.738µs,147.559517ms,23308
1571327156,781ns,780.453µs,136.00597ms,25151
1571327157,782ns,821.709µs,142.437387ms,24603
1571327158,785ns,1.57976ms,355.690413ms,12448
1571327159,774ns,848.865µs,186.433177ms,23890
1571327160,786ns,927.24µs,162.513711ms,21510
1571327161,779ns,759.99µs,135.826571ms,26173
1571327162,778ns,762.124µs,128.748933ms,26173
1571327163,782ns,1.236054ms,848.211744ms,20734
1571327164,804ns,2.772501ms,746.674495ms,5387
1571327165,782ns,921.272µs,191.09674ms,21025
1571327166,778ns,814.464µs,142.33664ms,24524
1571327167,770ns,814.118µs,137.200021ms,24569
1571327168,781ns,907.881µs,143.304291ms,22092
1571327169,772ns,862.67µs,145.800969ms,22823
1571327170,779ns,945.935µs,152.286216ms,13933
1571327171,828ns,1.557664ms,227.092562ms,146



--------- Bench heartbeat (Round 2) ----------
store 1 finish heartbeat, cost time: 7.637462195s
store 19 finish heartbeat, cost time: 7.899849567s
store 18 finish heartbeat, cost time: 7.936266795s
store 4 finish heartbeat, cost time: 8.027330119s
store 10 finish heartbeat, cost time: 8.064582617s
store 9 finish heartbeat, cost time: 8.092133678s
store 17 finish heartbeat, cost time: 8.212758254s
store 5 finish heartbeat, cost time: 8.245043888s
store 20 finish heartbeat, cost time: 8.719631093s
store 7 finish heartbeat, cost time: 8.962240003s
store 13 finish heartbeat, cost time: 9.09039731s
store 6 finish heartbeat, cost time: 9.29339005s
store 2 finish heartbeat, cost time: 9.353499612s
store 16 finish heartbeat, cost time: 9.480885412s
store 12 finish heartbeat, cost time: 9.598450296s
store 8 finish heartbeat, cost time: 9.739297485s
store 3 finish heartbeat, cost time: 9.784913961s
store 14 finish heartbeat, cost time: 9.785700283s
store 11 finish heartbeat, cost time: 9.793207832s
store 15 finish heartbeat, cost time: 9.794701415s

Summary:
  Total:	9.7948 secs.
  Slowest:	0.6721 secs.
  Fastest:	0.0000 secs.
  Average:	0.0002 secs.
  Stddev:	0.0057 secs.
  Requests/sec:	102097.3460

Response time histogram:
  0.0000 [1]	|
  0.0672 [999313]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.1344 [339]	|
  0.2016 [187]	|
  0.2688 [72]	|
  0.3361 [41]	|
  0.4033 [18]	|
  0.4705 [19]	|
  0.5377 [15]	|
  0.6049 [13]	|
  0.6721 [2]	|

Latency distribution:
  10% in 0.0000 secs.
  25% in 0.0000 secs.
  50% in 0.0000 secs.
  75% in 0.0000 secs.
  90% in 0.0000 secs.
  95% in 0.0000 secs.
  99% in 0.0000 secs.
  99.9% in 0.0329 secs.

Sample in one second (unix latency throughput):
UNIX-SECOND,MIN-LATENCY-MS,AVG-LATENCY-MS,MAX-LATENCY-MS,AVG-THROUGHPUT
1571327171,1.351µs,905.779µs,519.767383ms,14199
1571327172,805ns,1.497563ms,415.098309ms,11894
1571327173,796ns,1.454812ms,293.169369ms,14883
1571327174,828ns,3.321515ms,612.172878ms,6021
1571327175,802ns,2.639892ms,672.108266ms,8207
1571327176,789ns,175.226µs,339.219364ms,86945
1571327177,797ns,76.435µs,365.673545ms,256367
1571327178,845ns,139.559µs,359.332802ms,129828
1571327179,803ns,39.214µs,112.681827ms,375571
1571327180,1.013µs,130.202µs,109.811237ms,74195
1571327181,950ns,64.677µs,39.676263ms,21910



--------- Bench heartbeat (Round 3) ----------
store 10 finish heartbeat, cost time: 6.436020432s
store 1 finish heartbeat, cost time: 6.722400486s
store 5 finish heartbeat, cost time: 6.770380942s
store 9 finish heartbeat, cost time: 6.897113515s
store 17 finish heartbeat, cost time: 7.038131066s
store 19 finish heartbeat, cost time: 7.103957553s
store 18 finish heartbeat, cost time: 7.15742659s
store 13 finish heartbeat, cost time: 7.204494695s
store 4 finish heartbeat, cost time: 7.235026932s
store 15 finish heartbeat, cost time: 7.339481592s
store 20 finish heartbeat, cost time: 7.445103063s
store 7 finish heartbeat, cost time: 7.496123163s
store 11 finish heartbeat, cost time: 7.557848061s
store 6 finish heartbeat, cost time: 7.642173479s
store 14 finish heartbeat, cost time: 7.690908351s
store 12 finish heartbeat, cost time: 7.730360049s
store 16 finish heartbeat, cost time: 7.738155807s
store 3 finish heartbeat, cost time: 7.758237394s
store 8 finish heartbeat, cost time: 7.760809272s
store 2 finish heartbeat, cost time: 7.78686491s

Summary:
  Total:	7.7870 secs.
  Slowest:	0.9230 secs.
  Fastest:	0.0000 secs.
  Average:	0.0001 secs.
  Stddev:	0.0049 secs.
  Requests/sec:	128422.2551

Response time histogram:
  0.0000 [1]	|
  0.0923 [999526]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.1846 [341]	|
  0.2769 [91]	|
  0.3692 [30]	|
  0.4615 [7]	|
  0.5538 [16]	|
  0.6461 [5]	|
  0.7384 [1]	|
  0.8307 [0]	|
  0.9230 [2]	|

Latency distribution:
  10% in 0.0000 secs.
  25% in 0.0000 secs.
  50% in 0.0000 secs.
  75% in 0.0000 secs.
  90% in 0.0000 secs.
  95% in 0.0000 secs.
  99% in 0.0000 secs.
  99.9% in 0.0292 secs.

Sample in one second (unix latency throughput):
UNIX-SECOND,MIN-LATENCY-MS,AVG-LATENCY-MS,MAX-LATENCY-MS,AVG-THROUGHPUT
1571327181,790ns,718.352µs,923.025794ms,21936
1571327182,811ns,2.116386ms,575.663412ms,8005
1571327183,791ns,1.295264ms,304.157804ms,15346
1571327184,781ns,1.227551ms,280.918715ms,16183
1571327185,797ns,64.571µs,321.393424ms,253907
1571327186,897ns,131.986µs,598.345544ms,153543
1571327187,810ns,159.222µs,482.964175ms,117819
1571327188,820ns,35.152µs,107.004495ms,367189
1571327189,1.049µs,21.428µs,17.978842ms,46092

```

### CODE REVIEW

File: tools/heartbeat-bench/main.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
+// copyright 2019 pingcap, inc.
```
Comment (disksing) [inline tools/heartbeat-bench/main.go]: bad format

**Comment by rleungx** [comment]
It seems the output histogram is not well-formatted.

**Comment by nolouch** [comment]
@rleungx You are right, the problem may in `Report` package, may need to fix it in etcd.

File: Makefile [hunk 2]
```diff
@@ -69,6 +68,12 @@ pd-tso-bench:
 pd-recover: export GO111MODULE=on
 pd-recover:
 	CGO_ENABLED=0 go build -o bin/pd-recover tools/pd-recover/main.go
+pd-analysis: export GO111MODULE=on
+pd-analysis:
+	CGO_ENABLED=0 go build -gcflags '$(GCFLAGS)' -ldflags '$(LDFLAGS)' -o bin/pd-analysis tools/pd-analysis/main.go
+heartbeat-bench: export GO111MODULE=on
```
Comment (rleungx) [inline Makefile]: prefer to keep consistent with others like `pd-*`

**Pull request merged**